subgraphs/compound-v3/schema.graphql

```diff
@@ -0,0 +1,2020 @@
+# Subgraph Schema: Lending Protocol
+# Version: 3.0.0
+# See https://github.com/messari/subgraphs/blob/master/docs/SCHEMA.md for details
+
+enum Network {
+  ARBITRUM_ONE
+  ARWEAVE_MAINNET
+  AURORA
+  AVALANCHE
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum ProtocolType {
+  EXCHANGE
+  LENDING
+  YIELD
+  BRIDGE
+  GENERIC
+  # Will add more
+}
+
+type Token @entity {
+  " Smart contract address of the token "
+  id: Bytes!
+
+  " Name of the token, mirrored from the smart contract "
+  name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
+  symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
+  decimals: Int!
+
+  " Optional field to track the price of a token, mostly for caching purposes "
+  lastPriceUSD: BigDecimal
+
+  " Optional field to track the block number of the last token price "
+  lastPriceBlockNumber: BigInt
+
+  " The type of token the protocol creates for positions "
+  type: TokenType
+}
+
+enum RewardTokenType {
+  " For reward tokens awarded to LPs/lenders "
+  DEPOSIT
+
+  " For reward tokens awarded to borrowers of variable debt "
+  VARIABLE_BORROW
+
+  " For reward tokens awarded to borrowers of stable debt "
+  STABLE_BORROW
+
+  " For reward tokens awarded to stakers "
+  STAKE
+}
+
+type RewardToken @entity(immutable: true) {
+  " { Reward token type }-{ Smart contract address of the reward token } "
+  id: ID!
+
+  " Reference to the actual token "
+  token: Token!
+
+  " The type of the reward token "
+  type: RewardTokenType!
+}
+
+enum LendingType {
+  " Collateralized Debt Position (CDP) protocols have singular isolated positions created by users. We aggregate them to give a single view of a market "
+  CDP
+
+  " Pooled protocols pool all users assets into a single market "
+  POOLED
+}
+
+enum PermissionType {
+  " Only users that have been whitelisted can interact. e.g. Only approved institutions can borrow "
+  WHITELIST_ONLY
```

Comment:
In the `LendingProtocol` entity there are fields for `lenderPermissionType` and `borrowerPermissionType`. Does that cover what you are thinking?

Reply:
It is different - e.g. creation a new pool in compound [requires admin](https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/Comptroller.sol#L933), while theoretically [anyone can create a new pool in euler](https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/Comptroller.sol#L933). Maybe this is a type of permission we don't care about, then we can ignore this.

Protocols like goldfinch requires borrowers to be KYC'ed AND added to their whitelist (GOListed).

Reply:
To me whitelist implies KYC, but I guess that may not always be the case. We say in the comment: `Only approved institutions can borrow`

Reply:
As far as poolcreatorpermissions, I am not quite sure where I stand on necessity, I can add it though

Reply:
_We can move this discussion to a separate issue to not block the PR if you guys prefer._

Isn’t it the other way around? KYC implies whitelist, but whitelist doesn’t imply KYC. If some protocol decided to add a token or not based on a governance vote, it would be whitelist but not KYC.

About adding the field for `PoolCreatorPermissions`, I would do it. It feels strange to have it for general users but not for pool creators.

Reply:
added